packages/template-retail-react-app/app/components/experience/carousel/index.jsx

```diff
@@ -0,0 +1,191 @@
+        lg: `calc(${100 / mdCarouselSlidesToDisplay}%)`
+    }
+
+    const overflowXScroll = {
+        base: xsCarouselIndicators ? 'block' : 'none',
+        sm: xsCarouselIndicators ? 'block' : 'none',
+        md: smCarouselIndicators ? 'block' : 'none',
+        lg: mdCarouselIndicators ? 'block' : 'none'
+    }
+    const overflowXScrollValue = useBreakpointValue(overflowXScroll)
+
+    const components = regions[0]?.components || []
+    const itemCount = components.length
+
+    // Scroll the container left or right by 100%. Passing no args or `1`
+    // scrolls to the right, and passing `-1` scrolls left.
+    const scroll = useCallback((direction = 1) => {
+        scrollRef.current?.scrollBy({
+            top: 0,
+            left: (direction * window.innerWidth) / itemCount,
+            behavior: 'smooth'
+        })
+    })
+
+    // Our indicator implementation uses the scrollbar to show the context of the current
+    // item selected. Because MacOS hides scroll bars after they come to rest we need to
+    // force them to show. Please note that this feature only works on web-kit browsers,
+    // for all other brosers the scroller/indicator will be shown.
+    const css = `
+        .scroll-indicator::-webkit-scrollbar {
+            display:${overflowXScrollValue};
+            -webkit-appearance: none;
+            height: 8px;
+        }
+        .scroll-indicator::-webkit-scrollbar-thumb {
+            background-color: rgba(0, 0, 0, 0.5);
+        }
+    `
+
+    return (
+        <Box className={'carousel'} position="relative" data-testid="carousel">
+            <style>{css}</style>
+            <Stack className={'carousel-container'} data-testid="carousel-container" spacing={6}>
+                {textHeadline && (
+                    <Heading as="h2" fontSize="xl" textAlign="center">
+                        {textHeadline}
+                    </Heading>
+                )}
+
+                <Stack
+                    ref={scrollRef}
+                    className={'carousel-container-items scroll-indicator'}
+                    data-testid="carousel-container-items"
+                    direction="row"
+                    spacing={0}
+                    wrap="nowrap"
+                    overflowX="scroll"
+                    sx={{
+                        scrollPadding: 0,
+                        scrollSnapType: 'x mandatory',
+                        WebkitOverflowScrolling: 'touch'
+                    }}
+                >
+                    {components.map((component, index) => (
+                        <Box
+                            key={component?.id || index}
+                            flex="0 0 auto"
+                            width={itemWidth}
+                            style={{scrollSnapAlign: 'start'}}
+                        >
+                            <AspectRatio ratio={1}>
+                                <Component component={{...component, index}} />
```

Comment:
Good call. I did this because I was passing that **index** through and using it for the image provider as an **id** value. I've changed the usage for my example to no require the **id** value, instead it uses the components **id** as a seed to get the image. Not that it's important because that page is only an example and isn't going to be commited.